cmd/containerboot/main.go

```diff
@@ -150,8 +155,8 @@ func main() {
 		if err := ensureTunFile(cfg.Root); err != nil {
 			log.Fatalf("Unable to create tuntap device file: %v", err)
 		}
-		if cfg.ProxyTo != "" || cfg.Routes != nil || cfg.TailnetTargetIP != "" || cfg.TailnetTargetFQDN != "" {
-			if err := ensureIPForwarding(cfg.Root, cfg.ProxyTo, cfg.TailnetTargetIP, cfg.TailnetTargetFQDN, cfg.Routes); err != nil {
+		if cfg.ProxyTargetIP != "" || cfg.ProxyTargetDNSName != "" || cfg.Routes != nil || cfg.TailnetTargetIP != "" || cfg.TailnetTargetFQDN != "" {
+			if err := ensureIPForwarding(cfg.Root, cfg.ProxyTargetIP, cfg.ProxyTargetDNSName, cfg.TailnetTargetIP, cfg.TailnetTargetFQDN, cfg.Routes); err != nil {
```

Comment:
There's a race condition here. `ensureIPForwarding()` only runs once. It does a DNS lookup to determine whether IPv4 or IPv6 forwarding are needed. Subsequently, we perform periodic DNS lookups down in the `runLoop`. In theory, those could return IPv6 when the initial lookup didn't, at which point we may not have the right forwarding enabled.

It might be better to call `ensureIPForwarding` whenever the resolved addrs change.

Reply:
That's a good catch, I think we need to do the same for other proxy types too, so I'll do this in a separate PR

Reply:
I think(?) the problem is unique to the new DNS-based flow because of the DNS lookup. For static IPs, we know up-front whether it's IPv4 or IPv6.

Reply:
Yeah I was thinking tailnet IPs of the node that can change during the proxy Pod's run- however there is actually no case that I can think of where adding an IPv4/IPv6 tailnet address to a node that previously did not have it would result in reconfiguring firewall for an already running proxy Pod.

I've split the `ensureIPForwarding` and am only running it at the point when we know we need to configure firewall to avoid failing DNS resolution crashing containerboot upfront.
In practice, since this is mostly for kube, we likely don't need this at all because of https://github.com/tailscale/tailscale/blob/main/cmd/k8s-operator/deploy/manifests/proxy.yaml#L20